src/Microsoft.TestPlatform.CrossPlatEngine/Adapter/TestExecutionRecorder.cs

```diff
@@ -144,32 +119,15 @@ public void RecordEnd(TestCase testCase, TestOutcome outcome)
                 lock (this.testCaseEndStatusSyncObject)
                 {
                     isTestCaseEndAlreadySent = this.testCaseEndStatusMap.Contains(testCase.Id);
```

Comment:
Can we remove `testCaseEndStatusMap` too?

Reply:
Thought about , but wasn't sure so didn't take it up in this PR.